src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  deleteAnimal(animal) {
+    Animal.alive = Animal.alive.filter((item) => item !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  setBite() {
```

Comment:
BTW, why do you need `setBite` wrapper?
Just do everything you need inside `bite` method
That might be a reason you need to check it like this: `'hidden' in this`

This particular point isn't critical, but you might check it like this:
animal instanceof Herbivore && !animal.hidden

hidden always defined in Harbivere instance

Reply:
You are right )